### ISSUE TITLE
Stop flat-lining Mains as a stand in for planned spending from non-DP orgs

### ISSUE THREAD

**Issue by Stephen-ONeil** [TBS-EACPD/infobase#1439]
The flat-lining of estimates as a stand in for planned spending from non-DP orgs is an old practice now, I think done since InfoBase first began publishing planned spending. It is explained on the site via the following footnote:
>Planned expenditures for most organizations are sourced from Departmental Plans. Where [Planned Spending](http://dev.ea-ad.ca/master/index-eng.html#glossary/PLANNED_EXP) information is not available (notably for [Parliamentary Entities](http://dev.ea-ad.ca/master/index-eng.html#glossary/PARL_ORG), the Office of the Governor General's Secretary, the Canadian Security Intelligence Service, the Communication Security Establishment, the International Joint Commission (Canadian Section) and most [Crown Corporations](http://dev.ea-ad.ca/master/index-eng.html#glossary/CROWN_SA)), **for illustrative purposes only**, GC InfoBase presents the most recent Main Estimates figures for these organizations as being constant across the three planning years.

I don't think this decision was made by any current team members, and the consensus with the present team is that this is something we would like to stop doing.

In practice, the flat-lining of mains occurs in the program spending dataset itself, upstream. Questionable nature of the practice aside, this implementation has always been a persistent source of bugs. I think it was intended that the flat-lined Mains values would contribute to the government level roll up of planned spending, and _maybe_ appear in the report builder so that the totals matched (and maybe so that the flat-lined cases could be seen directly). Otherwise they weren't supposed to be presented in the org, CR, and program infographics. This meant everyone had to know about and write a check to identify and ignore the fake planned spending data whenever it was used... this has not always been done/done consistently (E.g. [this program](http://dev.ea-ad.ca/master/index-eng.html#infographic/crso/HC-BSU00/financial/.-.-(panel_key.-.-'crso_by_prog_exp)) is displaying flat-lined planned spending in it's CR level infographic, but the [relevant org level infographic](http://dev.ea-ad.ca/master/index-eng.html#infographic/dept/2/financial) has no planned information).

Side note, I've never actually known where the program level authorities for this come from. We don't otherwise publish authorities at the program level. 

Putting this practice to rest will require something along the lines of this:
  - [ ] upstream, drop the flat-lined Mains values in the program spending dataset
  - [ ] drop all planned spending from the gov level "welcome mat" panel
    - [ ] and planned FTEs. I don't think they perform the same flat lining, but they're still misleading at the gov level (clarify this)
    - [ ] optional, maybe show Mains, at least when there's a closed out year of authorities and PA is still 3/4 of a year off, as an extra year. Could do this at the gov level as well as the org levels in general.
      - [ ] to line up the FTE trends, highlight the "gap year" with a vertical dash on in the FTE panel
    - [ ] footnote explaining why the rolled up gov level isn't showing any planned information anymore
  - [ ] drop planned spending from the gov level "Authorities, Expenditures and Planned Spending" panel
    - [ ] maybe also consider dropping in-year estimates from this panel in general, or at least clarify when the value is non-final due to pending supps
    - [ ] maybe also reorder finance panels, this may or may not still be the most interesting/all encompassing panel once it loses planning information (in the government level case, DP org level version could keep the planning in it probably)
  - [ ] consider a new government level panel that shows the rolled up planned spending and planned FTE values, but importantly make it up-front about how planned data only covers a subset of all orgs
    - [ ] optional: work out an agreed upon methodology for estimating what share of actual FTEs and spending are represented by this planning universe (shares of most recent actual spending and FTEs by DP and non-DP orgs?), for context
  - [ ] optional, maybe as follow up: I'd like to split the datasets themselves and keep actual and planned program resources separate. They have different sources AND different universes (all programs vs DP org programs)

**Comment by nhamalainen-tbs** [comment]
CR level authorities (not program) are from Expenditures by Purpose, published with main estimates  (https://www.canada.ca/en/treasury-board-secretariat/services/planned-government-spending/government-expenditure-plan-main-estimates/2022-23-estimates/expenditures-purpose.html). The pipeline sources them from the Bluebooksbycoreresponsibility cube in the data warehouse.

**Comment by vseeto** [comment]
As far as I understand, we cannot publish no program level authorities as they do not exist. The Estimates provide authorities as the CR level only. CFMRS collects expenditures at the program level which can be rolled up to CRs through the EA data warehouse.

Are "in-year estimates" authorities? If yes, keep in the gov level. 

Agree with the optional follow up of splitting the actual and planned datasets. I suggest we discuss the option of separating FTE and financial data so that PA does not need DRR FTE data which is already issue "Consider decoupling program FTE resources from Spending resources #1323".

**Comment by Stephen-ONeil** [comment]
> CR level authorities (not program) are from Expenditures by Purpose, published with main estimates...

> As far as I understand, we cannot publish no program level authorities as they do not exist. The Estimates provide authorities as the CR level only. CFMRS collects expenditures at the program level which can be rolled up to CRs through the EA data warehouse.

Right, that's why I forgot about this, flat-lined mains for non-DP orgs are part of the "fake program" problem. By default (there may still be an exception or two), non-DP orgs do not actually have programs. Since they _do_ have CRs, their CR data is counted in the program level datasets for alignment purposes. To do this, their CR data is attributed to a fictitious child program under each CR (same name as that CR, InfoBase redirects to that CR's infographic if trying to visit the fictitious program). My confusion was how authorities were apparently being split down to the program level, but they _aren't_ because flat-lining is only being done for fake programs (actually CRs).

> Are "in-year estimates" authorities? If yes, keep in the gov level.

Yes, I meant the authorities coming from the current year, where, most likely, further supplementary estimates are expected and the authorities aren't final.
